server/src/main/java/org/schoellerfamily/pathfinderhelper/datamodel/SkillRank.java

```diff
@@ -0,0 +1,24 @@
+package org.schoellerfamily.pathfinderhelper.datamodel;
+
+public class SkillRank
+{
+	private Skill skill;
+	private Integer ranks;
+	public SkillRank(Skill skill)
+	{
+		this.skill = skill;
+		ranks = Integer.valueOf(1);
+	}
+	public Skill getSkill()
```

Comment:
For a persistent object, you have to have a setter for any field otherwise the object can't be populated when reading from the DB.

Also, I'm starting to not get how your schema is arranged.